src/librustdoc/clean/mod.rs

```diff
@@ -1286,11 +1287,12 @@ fn clean_qpath(hir_ty: &hir::Ty<'_>, cx: &mut DocContext<'_>) -> Type {
                 res: Res::Def(DefKind::Trait, trait_def),
                 segments: trait_segments.clean(cx),
             };
+            register_res(cx, trait_path.res);
```

Comment:
Does `hir::QPath::Resolved(None, _)` above also need `register_res` added?

Reply:
It uses `resolve_type`, which calls `register_res`. Also, it hasn't been changed in this PR.